slam_toolbox/src/slam_toolbox_common.cpp

```diff
@@ -446,10 +443,11 @@ bool SlamToolbox::shouldProcessScan(
   }
 
   // check moved enough, within 10% for correction error
-  const double dist2 = fabs((last_pose.GetX() - pose.GetX())*(last_pose.GetX() - 
-    pose.GetX()) + (last_pose.GetY() - pose.GetY())*
-    (last_pose.GetX() - pose.GetY()));
-  if(dist2 < 0.8 * min_dist2 || scan->header.seq < 5)
+  const auto sq_dist_to_last_accepted_pose = last_pose.SquaredDistance(pose);
+  static const auto dist_thresh_sq = pow(smapper_->getMapper()->getParamMinimumTravelDistance(), 2);
```

Comment:
Don't use `pow` for squaring, its much slower since `pow` http://www.cplusplus.com/reference/cmath/pow/ takes in a double so the process it uses to be general for floating point exponents is very slow.

Reply:
Could we keep the pow here for better readability? And shouldn't the value of a static const value be only computed once?

Reply:
I'd prefer if we tried to make it performant where simple to do so

Reply:
I get your point, but trying to make something a bit more performant was exactly the reason for the original bug.

Reply:
That's not an argument for why to make the software as performant as possible. The original error was due to a copy paste error, it had nothing to do with optimization, what you changed to isn't any less performant to compute the squared distance. Frankly, it didn't occur to me to use the karto methods in the common package. The computation additions for `pow2` vs `x*x` is well described and is a very basic optimization C++ engineers should be well aware of.